### ISSUE TITLE
Question about tT_loss

### ISSUE THREAD

**Issue by ccchobits** [Shark-NLP/DiffuSeq#18]
I am still confused about issue #17. The content of this issue has been duplicated as follow:

_There is a tT_loss term in the final loss:

[DiffuSeq/diffuseq/gaussian_diffusion.py](https://github.com/Shark-NLP/DiffuSeq/blob/901f8604d619c1923d69e57bd11894523309aab8/diffuseq/gaussian_diffusion.py#L629-L630)

Lines 629 to 630 in [901f860](https://github.com/Shark-NLP/DiffuSeq/commit/901f8604d619c1923d69e57bd11894523309aab8)

 out_mean, _, _ = self.q_mean_variance(x_start, th.LongTensor([self.num_timesteps - 1]).to(x_start.device)) 
 tT_loss =  mean_flat(out_mean ** 2) 
What is this? I cannot find it in the paper. And accroding to the code, the out_mean looks like the mean value of  as  from the diffusion forward procedure, and out_mean ** 2 should then be 
. Also, there seems no learnable params in the compute graph of tT_loss?

I wonder what is this term for, what is the meaning, and where it comes from?_

As for the comment from @yjc4, I think that term doesn't explain these issues, because obviously it has been dropped from step 1 to step 2 in equation (17) from the paper. Please provide me some hints. Thanks.

**Comment by summmeer** [comment]
Hi,
Yes, `tT_loss` does not pass the transformer layer, but there are still learnable params, i.e. the params of word embedding (from `x_start`). We can regard it as a kind of regularization, so in Eq.17 we move it to $R(||x_0||^2)$.

**Comment by Dawn-LX** [comment]
> I am still confused about issue #17. The content of this issue has been duplicated as follow:
> 
> _There is a tT_loss term in the final loss: [DiffuSeq/diffuseq/gaussian_diffusion.py](https://github.com/Shark-NLP/DiffuSeq/blob/901f8604d619c1923d69e57bd11894523309aab8/diffuseq/gaussian_diffusion.py#L629-L630) Lines 629 to 630 in [901f860](https://github.com/Shark-NLP/DiffuSeq/commit/901f8604d619c1923d69e57bd11894523309aab8)_
> 
> ```
>  out_mean, _, _ = self.q_mean_variance(x_start, th.LongTensor([self.num_timesteps - 1]).to(x_start.device)) 
>  tT_loss =  mean_flat(out_mean ** 2) 
> ```
> 
> _What is this? I cannot find it in the paper. And accroding to the code, the out_mean looks like the mean value of as from the diffusion forward procedure, and out_mean ** 2 should then be . Also, there seems no learnable params in the compute graph of tT_loss? I wonder what is this term for, what is the meaning, and where it comes from?_
> 
> As for the comment from @yjc4 in #17, I think that term doesn't explain these issues, because obviously it has been dropped from step 1 to step 2 in equation (17) from the paper. Please provide me some hints. Thanks.

just by the way, it's issue #16
